src/components/Calendar/hooks/useCalendarState.ts

```diff
@@ -68,16 +99,20 @@ export function useCalendarState(props: CalendarStateOptions): CalendarState {
         disabled,
         readOnly,
         value,
-        setValue,
+        setValue(date: DateTime) {
+            if (!disabled && !readOnly) {
+                const newValue = constrainValue(date, minValue, maxValue);
+                if (this.isCellUnavailable(newValue)) {
+                    return;
+                }
+                setValue(newValue.timeZone(inputTimeZone));
+            }
+        },
         timeZone,
         selectDate(date: DateTime, force = false) {
             if (!disabled) {
                 if (!readOnly && (force || this.mode === minMode)) {
-                    const newValue = constrainValue(date, minValue, maxValue);
-                    if (this.isCellUnavailable(newValue)) {
-                        return;
-                    }
-                    setValue(newValue);
+                    this.setValue(date.startOf(minMode));
```

Comment:
Why are you apply timeZone [there](https://github.com/gravity-ui/date-components/pull/36/files#diff-ef006b04b74e2f360e93bdff6025b71a1871482c28a3a9ddcb44c27138e2e7e2R89)

But not here?